### PR TITLE
Volume measurement tool 

### PR DESCRIPTION
**Pull request by Abizem**
Volume measurement implementation within ODM

### CODE REVIEW

File: app/static/app/js/components/Map.jsx [hunk 1]
```diff
@@ -7,6 +7,8 @@ import Leaflet from 'leaflet';
 import async from 'async';
 import 'leaflet-measure/dist/leaflet-measure.css';
 import 'leaflet-measure/dist/leaflet-measure';
+import 'leaflet-draw/dist/leaflet.draw.css';
+import 'leaflet-draw/dist/leaflet.draw';
```
Comment (pierotofy) [inline app/static/app/js/components/Map.jsx]: You might need to add `leaflet-draw` in `package.json` as well!

File: app/static/app/js/components/Map.jsx [hunk 2]
```diff
@@ -139,7 +141,7 @@ class Map extends React.Component {
                                     <i class="fa fa-cube"></i> 3D
                                 </button>`;
 
-            layer.bindPopup(popup);
+            //layer.bindPopup(popup);
```
Comment (pierotofy) [inline app/static/app/js/components/Map.jsx]: Is this disabling the popup permanently?

File: app/static/app/js/components/Map.jsx [hunk 3]
```diff
@@ -184,6 +186,78 @@ class Map extends React.Component {
...
+      e.layer.bindPopup('Volume: ' + paramList.toFixed(2) + 'Mètres Cubes (m3)');
+    });
+
+    this.map.on(L.Draw.Event.EDITED, function(e) {
+      e.layers.eachLayer(function(layer) {
+        var paramList = null;
+        $.ajax({
+            type: 'POST',
+            async: false,
+            url: '/api/projects/1/tasks/4/volume',
```
Comment (pierotofy) [inline app/static/app/js/components/Map.jsx]: Project ID and task ID are hard-coded here, perhaps replace them with:

`
const meta = layer[Symbol.for("meta")];
meta.task.project --> project id 
meta.task.id --> task id
`

File: app/static/app/js/components/Map.jsx [hunk 4]
```diff
@@ -184,6 +186,78 @@ class Map extends React.Component {
...
+      
+    this.map.on(L.Draw.Event.CREATED, function(e) {
+      e.layer.feature = {geometry: {type: 'Polygon'} }; 
+      featureGroup.addLayer(e.layer);
+
+      var paramList;
+        $.ajax({
+            type: 'POST',
+            async: false,
+            url: '/api/projects/4/tasks/7/volume',
```
Comment (pierotofy) [inline app/static/app/js/components/Map.jsx]: Same (see below)

File: app/static/app/js/components/Map.jsx (hunk 1 above)
Comment (Abizem) [reply app/static/app/js/components/Map.jsx]: Yes, of course

File: app/static/app/js/components/Map.jsx (hunk 2 above)
Comment (Abizem) [reply app/static/app/js/components/Map.jsx]: Maybe, we could find a way to disable this popup when the user start drawing the polygon for volume estimation!

File: app/static/app/js/components/Map.jsx (hunk 3 above)
Comment (Abizem) [reply app/static/app/js/components/Map.jsx]: right! i wwould test this,
const meta = layer[Symbol.for("meta")]; 
meta.task.project=project.id 
meta.task.id=task.id

File: app/static/app/js/components/Map.jsx (hunk 2 above)
Comment (pierotofy) [reply app/static/app/js/components/Map.jsx]: I think that would be a good way to do it, yes.

**Comment by pierotofy** [comment]
I merged these changes into the volume branch, where I will make some modifications to bring this feature to completion.

Thanks @Abizem!

**Pull request merged**